### ISSUE TITLE
Error when trying iwth MyEtherWallet

### ISSUE THREAD

**Issue by madvas** [jamespic/SpaceSuit#13]
I get following error when trying with https://www.myetherwallet.com/, choosing MetaMask as provider, with MetaMask disabled, SpaceSuit enabled, ledger unlocked. 

```
TransportError: Failed to sign with Ledger device: U2F TIMEOUT. Are you sure you are on a secure (SSL / HTTPS) connection?
```
The site is using https

**Comment by jamespic** [comment]
I suspect I know what's causing this - I think the workaround I put in for the Dai Dashboard has had some knock-on effects. If it's what I suspect, then I'd expect the issue to be intermittent, and for it also to affect MEW if you try to use the Ledger directly.

I'll see if I can smooth out these issues, and if not, either pull the Dai Dashboard workarounds entirely, or create some sort of list of which sites need which workarounds.

**Comment by madvas** [comment]
Cool thanks, nice extension by the way!

**Comment by madvas** [comment]
I tried with `0.2.8` still getting: `TransportStatusError: Ledger device: UNKNOWN_ERROR (0x6801). Are you sure you are on a secure (SSL / HTTPS) connection?`

**Comment by madvas** [comment]
oh it worked now on second try!

**Comment by jamespic** [comment]
Hmm.. Sounds like it's still flakier than I'd like.

I think the problem is multiple tabs talking to the Ledger at once. The Ledger doesn't deal at all well with concurrent requests, so needs to be protected by a mutex, but at the moment each tab has its own mutex.

There was a regression I introduced in 0.2.5 that meant that even tabs that weren't trying to talk to SpaceSuit would periodically poll `eth_accounts`, which exacerbated the problem. I fixed that regression in 0.2.7, along with a couple of other changes to make it less chatty.

Maybe some kind of cross-tab locking mechanism would work? Or maybe requests are only allowed on the visible tab? Or maybe I can find a way to make it even less chatty?

I'll have a think.
